### PR TITLE
Issue #59 - Allow changing connected wallet

### PR DESCRIPTION
**Pull request by orenku**

### CODE REVIEW

File: src/components/ConnectWallet/ConnectWalletModal.tsx [hunk 1]
```diff
@@ -75,6 +80,13 @@ export const DisconnectWalletModal: FC<DisconnectWalletModalProps> = ({
 }) => {
   const { address, connector: activeConnector } = useAccount()
   const { disconnect } = useDisconnect()
+  const chainId = Number(CHAIN_ID)
```
Comment (aquaticone) [inline src/components/ConnectWallet/ConnectWalletModal.tsx:83]: I think you'll want to merge main into your branch and use the new hook @vanillaHill created `useActiveNetwork` to get the active chain ID.

File: src/utils/chains.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
+/**
+ * List of all the networks supported by the Uniswap Interface
+ */
+export enum SupportedChainId {
```
Comment (aquaticone) [inline src/utils/chains.ts:4]: You can delete this file, should be handled in #4

File: src/utils/getExplorerLinks.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
+import { SupportedChainId } from './chains'
+
+const ETHERSCAN_PREFIXES: { [chainId: number]: string } = {
```
Comment (aquaticone) [inline src/utils/getExplorerLinks.ts:3]: You can grab block explorer URLs from the `chain` object from wagmi, returned by `useActiveNetwork` #4 

```tsx
declare type Chain = {
    /** ID in number form */
    id: number;
    /** Human-readable name */
    name: string;
    /** Internal network name */
    network: string;
    /** Currency used by chain */
    nativeCurrency: NativeCurrency;
    /** Collection of RPC endpoints */
    rpcUrls: {
        [key: string]: RpcUrls;
        default: RpcUrls;
    };
    /** Collection of block explorers */
    blockExplorers?: {
        [key: string]: BlockExplorer;
        default: BlockExplorer;
    };
    /** Collection of contracts */
    contracts?: {
        ensRegistry?: Contract;
        multicall3?: Contract;
    };
    /** Flag for test networks */
    testnet?: boolean;
};
```

File: src/utils/getExplorerLinks.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+  switch (type) {
+    case ExplorerDataType.TRANSACTION:
+      return `${prefix}/tx/${data}`
+
+    case ExplorerDataType.TOKEN:
+      return `${prefix}/token/${data}`
+
+    case ExplorerDataType.BLOCK:
+      // if (chainId === SupportedChainId.OPTIMISM || chainId === SupportedChainId.OPTIMISTIC_KOVAN) {
```
Comment (aquaticone) [inline src/utils/getExplorerLinks.ts:67]: Just delete any code we're not using

**Review by aquaticone — CHANGES_REQUESTED** [review]

File: src/components/ConnectWallet/ConnectWalletModal.tsx (hunk 1 above)
Comment (orenku) [reply src/components/ConnectWallet/ConnectWalletModal.tsx]: Fixed - Used Vanillla's implementation

File: src/utils/chains.ts (hunk 2 above)
Comment (orenku) [reply src/utils/chains.ts]: Deleted

File: src/utils/getExplorerLinks.ts (hunk 3 above)
Comment (orenku) [reply src/utils/getExplorerLinks.ts]: Great - Used that instead

File: src/utils/getExplorerLinks.ts (hunk 4 above)
Comment (orenku) [reply src/utils/getExplorerLinks.ts]: OK

**Comment by orenku** [comment]
@aquaticone Fixed the PR  - Kindly re-review

**Review by aquaticone — APPROVED** [review]
LGTM

**Pull request merged**